openjdk-integ-tests/src/test/java/org/conscrypt/javax/net/ssl/SSLSocketTest.java

```diff
@@ -2732,6 +2733,21 @@ private SecretKey newKey() {
         }
     }
 
+    // Tests that a socket will close cleanly even if it fails to create due to an
+    // internal IOException
+    @Test
+    public void test_SSLSocket_CloseCleanlyOnConstructorFailure() throws Exception {
+        TestSSLContext c = TestSSLContext.create();
+        Exception ex = null;
+        try {
+            SSLSocket client =
+                (SSLSocket) c.clientContext.getSocketFactory().createSocket(c.host, 1);
+        } catch (ConnectException e) {
+            ex = e;
+        }
+        assumeNoException(ex);
```

Comment:
I don't think we want this assume here.  ConnectionException should be thrown, and assuming it's not causes this test to report as skipped instead of passing.  I think you just want to do

```java
try {
  c.clientContext.getSocketFactory().createSocket(c.host, 1);
  fail();
} catch (ConnectException expected) {
  // Ignored
}
```

and skip doing anything with the ConnectException entirely.

Reply:
Done! Clearly, a bit too much copy/pasta before. Travis errored out on the OSX build before it even got to the tests, but I think they should still pass now.